plasmapy/tests/helpers/tests/sample_functions.py

```diff
@@ -0,0 +1,173 @@
+"""
+Sample functions and classes to be used for testing the test helper
+functionality.
+"""
+
+import astropy.units as u
+import numpy as np
+import warnings
+
+from typing import NoReturn
+
+
+class SampleException(Exception):
+    """A sample exception to be used for testing purposes."""
+
+    pass
+
+
+class SampleExceptionSubclass(SampleException):
+    """
+    The subclass of the `SampleException` to used for testing purposes.
+    """
+
+    pass
+
+
+class SampleWarning(Warning):
+    """A sample warning to be used for testing purposes."""
+
+    pass
+
+
+class SampleWarningSubclass(SampleWarning):
+    """
+    The subclass of the `SampleWarning` to be used for testing purposes.
+    """
+
+    pass
+
+
+def return_42() -> int:
+    """A sample function for testing purposes that returns ``42``."""
+
+    return 42
+
+
+def return_42_meters() -> u.Quantity:
+    """
+    A sample function for testing purposes that returns an
+    `~astropy.units.Quantity` with a value of ``42.0 m``.
+    """
+
+    return 42.0 * u.m
+
+
+def return_np_array(*args) -> np.array:
+    """A function to be used when testing `~numpy.array` instances."""
+
+    return np.array(args)
+
+
+def issue_warning() -> NoReturn:
+    """A sample function that issues a `SampleWarning`."""
+
+    warnings.warn("warning message", SampleWarning)
+
+
+def issue_warning_return_42() -> int:
+    """
+    A sample function for testing purposes that issues a `SampleWarning`
+    and returns ``42``.
+    """
+
+    warnings.warn("warning message", SampleWarning)
+    return 42
+
+
+def raise_exception():
+    """A sample function for testing purposes that raises a `SampleException`."""
+
+    raise SampleException("exception message")
+
+
+def sum_of_args_and_kwargs(arg1, arg2, *, kw1=None, kw2=None):
```

Comment:
```suggestion
def sum_of_args_and_kwargs(arg1, arg2, *, kw1, kw2):
```

Make keyword arguments required so you fail on the instantiation instead of the addition.